menas/src/main/scala/za/co/absa/enceladus/menas/auth/KerberosMenasAuthentication.scala

```diff
@@ -0,0 +1,137 @@
+import org.springframework.core.io.FileSystemResource
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
+import org.springframework.security.kerberos.authentication.KerberosServiceAuthenticationProvider
+import org.springframework.security.kerberos.authentication.sun.SunJaasKerberosTicketValidator
+import org.springframework.security.kerberos.client.config.SunJaasKrb5LoginConfig
+import org.springframework.security.kerberos.client.ldap.KerberosLdapContextSource
+import org.springframework.security.kerberos.web.authentication.SpnegoAuthenticationProcessingFilter
+import org.springframework.security.kerberos.web.authentication.SpnegoEntryPoint
+import org.springframework.security.ldap.authentication.ad.ActiveDirectoryLdapAuthenticationProvider
+import org.springframework.security.ldap.search.FilterBasedLdapUserSearch
+import org.springframework.security.ldap.userdetails.LdapUserDetailsMapper
+import org.springframework.security.ldap.userdetails.LdapUserDetailsService
+import org.springframework.stereotype.Component
+
+import za.co.absa.enceladus.menas.ActiveDirectoryLdapAuthoritiesPopulator
+
+@Component("kerberosMenasAuthentication")
+class KerberosMenasAuthentication() extends MenasAuthentication with InitializingBean {
+  @Value("${za.co.absa.enceladus.menas.auth.ad.domain:}")
+  val adDomain: String = ""
+  @Value("${za.co.absa.enceladus.menas.auth.ad.server:}")
+  val adServer: String = ""
+  @Value("${za.co.absa.enceladus.menas.auth.sysuser.principal:}")
+  val servicePrincipal: String = ""
+  @Value("${za.co.absa.enceladus.menas.auth.sysuser.keytab.location:}")
+  val keytabLocation: String = ""
+  @Value("${za.co.absa.enceladus.menas.auth.ldap.search.base:}")
+  val ldapSearchBase: String = ""
+  @Value("${za.co.absa.enceladus.menas.auth.ldap.search.filter:}")
+  val ldapSearchFilter: String = ""
+  @Value("${za.co.absa.enceladus.menas.auth.kerberos.debug:false}")
+  val kerberosDebug: Boolean = false
+
+  lazy val requiredParameters = Seq((adDomain, "za.co.absa.enceladus.menas.auth.ad.domain"),
+    (adServer, "za.co.absa.enceladus.menas.auth.ad.server"),
+    (servicePrincipal, "za.co.absa.enceladus.menas.auth.sysuser.principal"),
+    (keytabLocation, "za.co.absa.enceladus.menas.auth.sysuser.keytab.location"),
+    (ldapSearchBase, "za.co.absa.enceladus.menas.auth.ldap.search.base"),
+    (ldapSearchFilter, "za.co.absa.enceladus.menas.auth.ldap.search.filter"))
+
+  override def afterPropertiesSet() {
+    System.setProperty("javax.net.debug", kerberosDebug.toString)
+    System.setProperty("sun.security.krb5.debug", kerberosDebug.toString)
+  }
+
+  private def validateParam(param: String, paramName: String) = {
+    if (param.isEmpty()) throw new IllegalArgumentException(s"$paramName has to be configured in order to use kerberos Menas authentication")
```

Comment:
```suggestion
    if (param.isEmpty()) {
      throw new IllegalArgumentException(s"$paramName has to be configured in order to use kerberos Menas authentication")
    }
```
Subjectively better.